src/js/directives/gravatarList.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+var _ = require('lodash');
+module.exports = function(app) {
+
+  app.directive('gravatarList', function() {
+    return {
+      restrict: 'E',
+      scope: {
+        discussion: '=',
+      },
+      templateUrl: 'templates/directives/gravatarList.html',
+      link: function(scope, element, attrs) {
+        scope.$watch('discussion', function(discussion) {
+          // TODO $watch doesn't fire on reply update
```

Comment:
You can set the `objectEquality` parameter of `$watch` to `true`, see the [docs](https://docs.angularjs.org/api/ng/type/$rootScope.Scope).